src/store/actions.js

```diff
@@ -1,15 +1,22 @@
-import { ADD_TASK, DELETE_TASK } from "@store/types";
+import { ADD_TASK, DELETE_TASK, SET_TASK_DONE } from "@store/types";
 
 export const addTask = (title) => ({
   type: ADD_TASK,
   payload: {
     id: Date.now().toString(),
     title: title,
     createdDate: new Date(),
+    completedDate: null,
+    isDone: false,
   },
 });
 
 export const deleteTask = (taskId) => ({
   type: DELETE_TASK,
   payload: taskId,
 });
+
+export const setTaskDone = (taskId) => ({
+  type: SET_TASK_DONE,
```

Comment:
SET_TASK_DONE seems a bit confusing with useState setter. you may rename it to this 
```suggestion
  type: Complete_TASK,
```